libraries/AP_HAL/UARTDriver.h

```diff
@@ -42,6 +47,10 @@ class AP_HAL::UARTDriver : public AP_HAL::BetterStream {
      */
     void begin_locked(uint32_t baud, uint16_t rxSpace, uint16_t txSpace, uint32_t write_key);
 
+#if HAL_UART_DEBUG_LOGGING_ENABLED
+    void set_instance(uint8_t _instance) { instance = _instance; }
+#endif  // HAL_UART_DEBUG_LOGGING_ENABLED
```

Comment:
How does this work for e.g. network and CAN ports? Instance should be > 9, but it's unclear if they will even be set up properly or the functions called. Does anything bad happen if you try?